src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineLinuxLauncher.java

```diff
@@ -119,9 +119,9 @@ protected void launch(ComputeEngineComputer computer, TaskListener listener, Ins
             conn = cleanupConn;
 
             SCPClient scp = conn.createSCPClient();
-            String tmpDir = "/tmp";
-            logInfo(computer, listener, "Copying agent.jar to: " + tmpDir);
-            scp.put(Jenkins.get().getJnlpJars("agent.jar").readFully(), "agent.jar", tmpDir);
+            String jenkinsDir = node.getRemoteFS();
```

Comment:
This code is copy/pasted between launchers..

Cannot you extract it somewhere? Or its huge change?

Reply:
Yes, it will have to be part of a larger refactoring effort.